tests/resources/serializers/test_marcxml_serializer.py

```diff
@@ -0,0 +1,93 @@
+
+from collections import Iterable
+
+import pytest
+from dojson.contrib.marc21.utils import GroupableOrderedDict, create_record
+
+from invenio_rdm_records.resources.serializers.marcxml import MARCXMLSerializer
+
+
+@pytest.fixture(scope="function")
+def updated_minimal_record(minimal_record):
+    """Update fields (done after record create) for MARCXML serializer."""
+    minimal_record["access"]["status"] = "open"
+    for creator in minimal_record["metadata"]["creators"]:
+        name = creator["person_or_org"].get("name")
+        if not name:
+            creator["person_or_org"]["name"] = "Name"
+
+    return minimal_record
+
+
+@pytest.fixture(scope="function")
+def updated_full_record(full_record):
+    """Update fields (done after record create) for MARCXML serializer."""
+    full_record["access"]["status"] = "embargoed"
+
+    return full_record
+
+
+def test_marcxml_serializer_minimal_record(running_app, updated_minimal_record):
+    """Test serializer for MARCXML"""
+    serialized_record = {}
+    expected_data = {}
+
+    serializer = MARCXMLSerializer()
+    serialized_record = create_record(
+        serializer.serialize_object(updated_minimal_record)
+    )
+    expected_data = '<record xmlns="http://www.loc.gov/MARC21/slim"><datafield tag="245" ind1="a" ind2=" "><subfield>A Romans story</subfield></datafield><controlfield tag="540">info:eu-repo/semantics/openAccess</controlfield><datafield tag="260" ind1="c" ind2=" "><subfield>2020-06-01</subfield></datafield><datafield tag="100" ind1="a" ind2=" "><subfield>Name</subfield></datafield><datafield tag="100" ind1="a" ind2=" "><subfield>Troy Inc.</subfield></datafield><controlfield tag="901">info:eu-repo/semantic/other</controlfield><datafield tag="260" ind1="b" ind2=" "><subfield>Acme Inc</subfield></datafield></record>'
+    expected_data = create_record(expected_data)
+
+    record1 = flatten(record_to_string_list(serialized_record))
+    record2 = flatten(record_to_string_list(expected_data))
+
+    record1 = set(record1)
+    record2 = set(record2)
+
+    assert record1 == record2
+
+
+def record_to_string_list(record):
+    """Recursively unnest all elements from GroupableOrderedDict"""
+    if isinstance(record, str):
+        return record
+    if isinstance(record, GroupableOrderedDict):
+        return record_to_string_list(record.values())
+    elements = []
+    for rec in record:
+        elements.append(record_to_string_list(rec))
+    return elements
+
+
+def flatten(elements):
+    """Flattens a n-dimensional irregular list"""
+    if isinstance(elements, Iterable) and not isinstance(elements, str):
+        return [a for i in elements for a in flatten(i)]
+    else:
+        return [elements]
+
+
+def test_marcxml_serializer_full_record(running_app, updated_full_record):
+    """Test serializer for MARCXML"""
+    serializer = MARCXMLSerializer()
+    serialized_record = create_record(serializer.serialize_object(updated_full_record))
```

Comment:
Do we need the `create_record` call here? Is that function being called also when we serialize it via the REST API? Could we test exactly the output that the serializer is giving back from the REST API point of view? i.e when we call `/records/<pid_value>/export/marcxml`?

Reply:
_Do we need the create_record call here?_
I will try testing without and let you know!

 _Is that function being called also when we serialize it via the REST API?_
 I don't think so, so it's quite probable I can remove it in the test
 
 _Could we test exactly the output that the serializer is giving back from the REST API point of view? i.e when we call /records/<pid_value>/export/marcxml?_
 Do you mean making a GET request to the running server instance?
 In which case I think it should be possible, although the current test already tests effectively the same (by calling the function that serializes the record and that is then returned by the request)